sources/packages/backend/apps/api/src/route-controllers/student-appeal/student-appeal.controller.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable, NotFoundException } from "@nestjs/common";
+import {
+  StudentAppealAPIOutDTO,
+  DetailedStudentAppealAPIOutDTO,
+  StudentAppealRequestAPIOutDTO,
+  DetailedStudentAppealRequestAPIOutDTO,
+} from "./models/student-appeal.dto";
+import { getUserFullName } from "../../utilities";
+import { StudentAppealService } from "../../services";
+
+@Injectable()
+export class StudentAppealControllerService {
+  constructor(private readonly studentAppealService: StudentAppealService) {}
+
+  /**
+   * Get the student appeal and its requests.
+   * @param appealId appeal id to be retrieved.
+   * @param studentId student id.
+   * @returns the student appeal and its requests.
+   */
+  async getStudentAppealWithRequest(
+    appealId: number,
+    studentId?: number,
+    options?: {
+      assessDetails?: boolean;
+    },
+  ): Promise<StudentAppealAPIOutDTO | DetailedStudentAppealAPIOutDTO> {
+    const studentAppeal =
+      await this.studentAppealService.getAppealAndRequestsById(
+        appealId,
+        studentId,
+      );
+    if (!studentAppeal) {
+      throw new NotFoundException("Not able to find the student appeal.");
+    }
+
+    return {
+      id: studentAppeal.id,
+      submittedDate: studentAppeal.submittedDate,
+      status: studentAppeal.status,
+      appealRequests: studentAppeal.appealRequests.map((appealRequest) => {
+        let requests = {
+          id: appealRequest.id,
+          appealStatus: appealRequest.appealStatus,
+          submittedData: appealRequest.submittedData,
+          submittedFormName: appealRequest.submittedFormName,
+        } as StudentAppealRequestAPIOutDTO;
+        if (options?.assessDetails) {
```

Comment:
If the idea is to keep both DTOs we can use the approach here also to ensure the right type return in the future also.
https://github.com/bcgov/SIMS/pull/1924#discussion_r1191476220

Reply:
I got what you are saying, actually, when I implemented https://github.com/bcgov/SIMS/pull/1926#discussion_r1191532022   there is no need of 2 DTO, 
![image](https://github.com/bcgov/SIMS/assets/77353155/9f8b8c7a-4fc8-4d17-a0cb-57c752680587)